src/util/lang_class.js

```diff
@@ -2,6 +2,23 @@
 
   var slice = Array.prototype.slice, emptyFunction = function() { },
 
+      makePropertyDescriptor = function(value) {
```

Comment:
The `makePropertyDescriptor ` creates a descriptor object for properties that were added to an object without descriptor.

Let's take the following example:
```javascript
var source = {};
source.foo = 'bar';
var descriptor = Object.getOwnPropertyDescriptor(source, 'foo');
```
Here, the `descriptor` object is `undefined`, because the property `foo` was not set using a descriptor. Because `Object.defineProperty` requires a descriptor, we need to construct one ourselves. You can read more about the individual arguments of a descriptor in [this](http://patrickdelancy.com/2012/09/property-descriptors-in-javascript/) slightly outdated blog, which I Googled.

We can write the exact same code (behaviour-wise) like this:
```javascript
var source = {};
Object.defineProperty(source, 'foo', {
  value: 'bar', //initial value
  writable: true, //allow source.foo = 'foobar'
  enumerable: true, //('foo' in source) will be true
  configurable: true //allow the descriptor to be changed later
})
```
The advantage to using descriptors can now be seen, as we're able to completely control how this property must behave on an engine level. If we want the value of the property to be constant, we would just give the `writable` argument a value of `false`.